lib/daru/io/importers/csv.rb

```diff
@@ -5,15 +5,53 @@ module IO
     module Importers
       module CSV
         class << self
-          def load(str='CSV#load')
-            CSVHelper.manipulate str
+          def load(path, opts={})
+            daru_options, opts = CSVHelper.prepare_opts opts
+            # Preprocess headers for detecting and correcting repetition in
+            # case the :headers option is not specified.
+            hsh =
+              if opts[:headers]
+                CSVHelper.hash_with_headers(path, opts)
+              else
+                CSVHelper
+                  .hash_without_headers(path, opts)
+                  .tap { |hash| daru_options[:order] = hash.keys }
+              end
+            Daru::DataFrame.new(hsh,daru_options)
           end
         end
       end
       module CSVHelper
         class << self
-          def manipulate(str)
-            "CSVHelper#manipulate called by #{str}"
+          OPT_KEYS = %i[clone order index name].freeze
+
+          def prepare_opts(opts)
+            opts[:col_sep]           ||= ','
+            opts[:converters]        ||= :numeric
```

Comment:
I believe you can have Ruby 2.1 minimum as a target version, therefore finally switch to keyword arguments instead of option hashes.